src/api/app/controllers/webui/projects/pulse_controller.rb

```diff
@@ -37,14 +37,22 @@ def show
       private
 
       def set_range
-        @range = params[:range] == 'month' ? 'month' : 'week'
-
-        @date_range = case @range
-                      when 'month'
-                        1.month.ago..Date.tomorrow
-                      else
-                        1.week.ago..Date.tomorrow
-                      end
+        default_from = 1.week.ago
+        default_to = 0.days.ago
+
+        params[:from] ||= default_from.strftime('%Y-%m-%d')
+        params[:to] ||= default_to.strftime('%Y-%m-%d')
+
+        @date_range_from = DateTime.parse(params[:from])
+        @date_range_to = DateTime.parse(params[:to])
+
+        if @date_range_to.to_i < @date_range_from.to_i
+          flash.now[:error] = 'From newer than To, using default time range'
+          @date_range_from = default_from
+          @date_range_to = default_to
```

Comment:
Nitpick: these lines are not tested. It wouldn't be much work to add a spec for this case, I guess.